### ISSUE TITLE
Any plans to implement QUIC protocol?

### ISSUE THREAD

**Issue by tysonrm** [second-state/wasmedge_wasi_socket#19]

**Comment by tysonrm** [comment]
I have a JS microservice framework that implements deployment and language independence without the need for distribution and obviates the need for deployment automation (called federated microservices or microservice libraries) . It is based on module federation, hexagonal architecture and wasm. The complexity of distributed systems is the main reason microservice projects fail and why many orgs choose not to implement. 

Anyway, the framework will operate like a uniform substrate or fabric that transparently integrates and dynamically deploys and runtime binds components. Support for fast streaming and manipulation of streams (e.g. AI interference) is critical. I am comfortable with async I/O capabilities of Node but need the speed and non-blocking behavior of QUIC. Any provider can plug into the service mesh, which runs in process with the federation host (not as a sidecar). Support for QUIC has been dark launced in Node 16 (requires recompilation with expiremental flag). So exploring possibilities. Thanks, Michael. Love your work!

**Comment by tysonrm** [comment]
I'll leave this open for now, if that's okay, to track development of a Rust API for Node's Quic implementation, which I plan to develop in the near future.

**Comment by juntao** [comment]
Yes, I think WasmEdge can be embedded into your JS framework as a runtime for 3rd party plug-ins. Networking can be done in the JS host or in WasmEdge.
